bundles/org.eclipse.ui.workbench/Eclipse UI/org/eclipse/ui/internal/WorkbenchPlugin.java

```diff
@@ -1048,6 +1060,9 @@ public void stop(BundleContext context) throws Exception {
 			testableTracker.close();
 			testableTracker = null;
 		}
+		// Store recently used new page shortcuts to preferences
+		Set<String> selectedFromOther = RecentNewWizardSelection.getInstance().getSelectedFromOther();
```

Comment:
This code doesn't belong to the bundle activator, it is an internal implementation detail of the RecentNewWizardsPreferenceManager and should be done there. What yuo could do is to call RecentNewWizardsPreferenceManager.shutdown() here.